### ISSUE TITLE
ZSH exit code not passed to the prompt

### ISSUE THREAD

**Issue by jpwallace22** [warpdotdev/Warp#3034]
### Describe the bug

When I try to pass the exit code to my prompt using `zsh` and `$?` it always returns 0. Even if I run a function that returns 1. It will still be a 0. 

### To Reproduce

```shell
# in ~/.zshrc

show_fail() {
    if [ $? = 1 ]; then echo "FAILED"; fi;
}

PROMPT='$(show_fail)'

function forced_fail() {
  return 1
}
```
```shell
$ forced_fail
```

### Expected behavior

Expect to see: "FAILED" displayed on the prompt
What I see:  Nothing. The command failed (turns red) but does not show "FAILED"

## In Terminal
![Screen Shot 2023-05-06 at 10 41 22 AM](https://user-images.githubusercontent.com/93415734/236639017-0b9f81f9-2e39-4762-8dfa-0806cc60fd87.png)

## in Warp
![Screen Shot 2023-05-06 at 10 42 11 AM](https://user-images.githubusercontent.com/93415734/236639045-59d101dd-752c-4263-9eef-2d81a669ca59.png)

### Operating System

MacOS

### Operating System and Version

Monterey 12.6.3

### Shell Version

zsh 5.8.1 (x86_64-apple-darwin21.0)

### Warp Version

v0.2023.05.02.08.03.stable_00

### Does this block you from using Warp daily?

No

### Is this a Warp specific issue? (i.e. does it happen in Terminal, iTerm, Kitty, etc.)

Yes, this I confirmed this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b8107fdf-ba31-488d-b103-d271c89cac3e

None

**Comment by dannyneira** [comment]
Thanks for letting us know @jpwallace22, I think this may have something to do with Warp showing red blocks for anything other than return 0, or something to that effect and not necessarily running the echo after returning exit code 1. We'll post any updates on this thread.

Please add a :+1: to the original post at the top or comment with your details to signal that you're also having the same issue, and then hit the subscribe button if you'd like to be notified.

**Comment by jpwallace22** [comment]
> I think this may have something to do with Warp showing red blocks for anything other than return 0

@dannyneira Yeah, sounds like it has to do with the red blocks. I was running a virtual environment in Warp today and noticed that it doesn't show the red blocks but it DOES show my custom fail icon on the prompt whenever it failed.
